pkg/package/template.go

```diff
@@ -0,0 +1,205 @@
+package _package
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+type templateSpec struct {
+	Description string   `yaml:"description"`
+	Packages    []string `yaml:"packages"`
+}
+
+type TemplateManifest struct {
+	Name            string                 `yaml:"name"`
+	Annotations     map[string]string      `yaml:"annotations,omitempty"`
+	Description     string                 `yaml:"description"`
+	Commands        []Command              `yaml:"commands"`
+	Overlay         map[string]string      `yaml:"overlay,omitempty"`
+	VariableSchemas map[string]interface{} `yaml:"variables,omitempty"`
+}
+
+func Template(name, file string) error {
```

Comment:
This function belongs in the cmd package.  functions like mergePackages can stay in this package.